<big>今天开始用*jupyter notebook*的形式来重写一下[python challenge]("http://www.pythonchallenge.com")这个古老的`python`解谜网站的解法思路<br>
<br>
当然，我不是那个可以完全独立解答出来的大神，也是收集了八方的解题经验来结合自己的理解编写出来`python`程序的</big>

---

#### 第<del>1</del>0题的地址在[warming up](http://www.pythonchallenge.com/pc/def/0.html "#0")
* <img src="http://www.pythonchallenge.com/pc/def/calc.jpg" alt="calc.jpg" width="30%" height="30%"/>
* 图像是一个老式电视机，上面是风景的画面，画面中间有三个数字(2, 3, 8)
* 图像下面有一行提示：
> try to change the URL address.<br>
> 尝试改变URL地址

第一个想法（结合标题warming up）就是把URL地址的[`0.html`][0]改成[`1.html`][1]<br>
结果如下：
> 2**38 is much much larger.

[0]: http://www.pythonchallenge.com/pc/def/0.html
[1]: http://www.pythonchallenge.com/pc/def/1.html

所以这个也还是一个题目，应该跟画面中的数字有关<br>
尝试[`238.html`][2]：
> No... the 38 is a little bit above the 2...

所以应该是2的38次方——[`2**38.html`][3]：
> give the answer, not the question.

#### 至此就豁然开朗了！
---

[2]: http://www.pythonchallenge.com/pc/def/238.html
[3]: http://www.pythonchallenge.com/pc/def/2**38.html

In [1]:
print(2 ** 38)

274877906944


URL改成[`274877906944.html`][4]，页面跳转到了[`map.html`][5]<br>
这回应该对了吧。还真是warming up :)

[4]: http://www.pythonchallenge.com/pc/def/274877906944.html
[5]: http://www.pythonchallenge.com/pc/def/map.html

### 总结：这个古老的python挑战应该是个网址更改游戏，每一关都是一个网页，根据页面提示得到下一关的网址。
<small>事实上，这一关的图片名称叫`calc.jpg`，这也是一个提示，以后的题目里面会用得到这种信息的。</small>
###### 本题代码地址[`0.ipynb`](https://github.com/StevenPZChan/pythonchallenge/blob/notebook/nbfiles/0.ipynb)